atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -49,6 +49,19 @@
     @GuardedBy("this")
     PersistentLockId lockId;
 
+    /* This is used to prevent the following error case:
+     * 1. Sweep thread 1 grabs the lock for deleting stuff
+     * 2. Sweep thread 2 grabs the lock for deleting stuff
+     * 3. Sweep thread 1 releases the lock
+     * 4. A backup starts
+     * 5. Sweep thread 2 releases the lock
+     *
+     * A backup taken between steps 4 and 5 is not guaranteed to be consistent, because sweep may be deleting data
+     * that it relies on.
+     */
+    @GuardedBy("this")
+    int referenceCount = 0;
```

Comment:
I *think* this is safe, and doesn't need to be Atomic because it's only used in synchronised methods. Does that make sense? (@jeremyk-91 @sandorw  for SA in case this commit comment doesn't show up in the set of commits they view)

Reply:
For the sake of the visibility of changes, I believe this needs to be either `volatile` or an `AtomicInt`

Reply:
Looks like I was actually wrong here - `synchronized` access to `referenceCount` offers similar visibility guarantees to `volatile`.

Reply:
For reference, see comments on https://stackoverflow.com/questions/1850270/memory-effects-of-synchronization-in-java